### PR TITLE
test coverage for internal/webhooks

### PR DESCRIPTION
**Pull request by gflarity**
#### What type of PR is this?

Testing

#### What this PR does / why we need it:

Increased test coverage for internal/webhooks

#### Special notes for your reviewer:

Prompt:

Please create an idiomatic golang unit tests for this specific directory/package. Herea are the instructions: 

* Start by revieiwng the state of test coverage with golang tooling. 
* Be sure to document the fields in the anonymous test structs and how they'll be used below. 
* Each Test* function should also have some concise documentation as well. Each test case should have a brief explanation and expectation explaine. This should be right before the go struct definition for that test if it's table driven tests. 
* Each test should have a name field for cross referencing failures. 
* Rather than having a description field, the description should just be in comments above the name in the struct declaration. 
* Don't forget to run the tests and confirm they're working, but focus on just testing this package as others might have issues. 
* Please avoid creating skipped tests. 
* Refactor code to make it more testable if necessary. However keep these refactors to a minimum and as simple as possible. 
* Specifically hardcoded constants etc are good candidates to be refactor out of testable code. As are interface that can be mocked in a straight forward way. However you must remember to ensure backwards compatibility and new functions rather than changing the signature of public functions. Avoid making these test helpers public though. Only make such refactors when there's a good return on investment. 
* Please ensure coverage is as good as possible using golang tooling. 
* Avoid creating table driven tests when the test function just has a single test in it. 
* Avoid test that don't really test anything. 
* Avoid describing tests as edge cases. 
* Avoid tests for command line arguments. Try to keep test cases unique and adding value (avoid duplicate test cases). 

#### Does this PR introduce a API change?
```release-note
None
```

### CODE REVIEW

File: operator/internal/webhook/register_test.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+			namespace:          "",
+			serviceAccountName: "test-sa",
+			expected:           "system:serviceaccount::test-sa",
+		},
+		{
+			name:               "empty service account",
+			namespace:          "default",
+			serviceAccountName: "",
+			expected:           "system:serviceaccount:default:",
+		},
```
Comment (renormalize) [inline operator/internal/webhook/register_test.go]: Not sure if these are meaningful. We can just get rid of these.
```suggestion
```

File: operator/internal/webhook/register_test.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+	// Ensure env var is not set
+	originalEnv := os.Getenv(constants.EnvVarServiceAccountName)
+	err := os.Unsetenv(constants.EnvVarServiceAccountName)
+	require.NoError(t, err)
+	defer func() {
+		if originalEnv != "" {
+			os.Setenv(constants.EnvVarServiceAccountName, originalEnv)
+		}
+	}()
```
Comment (renormalize) [inline operator/internal/webhook/register_test.go]: I feel this handling is not required, because this code only runs in tests on a developer's machine/CI.
```suggestion
	// Ensure env var is not set
	err := os.Unsetenv(constants.EnvVarServiceAccountName)
	require.NoError(t, err)
```

File: operator/internal/webhook/register_test.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	originalEnv := os.Getenv(constants.EnvVarServiceAccountName)
+	os.Setenv(constants.EnvVarServiceAccountName, "test-sa")
+	defer func() {
+		if originalEnv != "" {
+			os.Setenv(constants.EnvVarServiceAccountName, originalEnv)
+		} else {
+			err := os.Unsetenv(constants.EnvVarServiceAccountName)
+			require.NoError(t, err)
+		}
+	}()
```
Comment (renormalize) [inline operator/internal/webhook/register_test.go]: https://github.com/ai-dynamo/grove/pull/230#discussion_r2499312836

File: operator/internal/webhook/register_test.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+	// Create a real webhook server
+	server := webhook.NewServer(webhook.Options{
+		Port: 9443,
+	})
+	mgr.WebhookServer = server
+
+	// Set env var
+	originalEnv := os.Getenv(constants.EnvVarServiceAccountName)
+	os.Setenv(constants.EnvVarServiceAccountName, "test-sa")
```
Comment (renormalize) [inline operator/internal/webhook/register_test.go]: We can let the testing framework handle the setting, and resetting of the env var to the original value.

```suggestion
	t.Setenv(constants.EnvVarServiceAccountName, "test-sa")
```

File: operator/internal/webhook/register_test.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+	defer func() {
+		if originalEnv != "" {
+			os.Setenv(constants.EnvVarServiceAccountName, originalEnv)
+		} else {
+			err := os.Unsetenv(constants.EnvVarServiceAccountName)
+			require.NoError(t, err)
+		}
+	}()
```
Comment (renormalize) [inline operator/internal/webhook/register_test.go]: https://github.com/ai-dynamo/grove/pull/230#discussion_r2499329029

File: operator/internal/webhook/admission/pcs/validation/util_test.go [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+}
+
+// Helper functions
+func stringPtr(s string) *string {
+	return &s
+}
+
+func intPtr(i int) *int {
+	return &i
+}
```
Comment (renormalize) [inline operator/internal/webhook/admission/pcs/validation/util_test.go]: Use the generic function `ptr.To()` instead (`"k8s.io/utils/ptr"`). Cleaner, and we've already used that in quite a few places as well.

File: operator/internal/webhook/admission/pcs/validation/handler_test.go [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+				WithPodCliqueTemplateSpec(
+					testutils.NewPodCliqueTemplateSpecBuilder("test").
+						WithReplicas(1).
+						WithRoleName("test-role").
+						WithMinAvailable(1).
+						Build()).
+				Build(),
+			expectError:   true,
+			errorContains: "field is immutable",
+		},
```
Comment (renormalize) [inline operator/internal/webhook/admission/pcs/validation/handler_test.go]: Logic in `validatePodCliqueUpdate` and `validatePodCliqueScalingGroupConfigsUpdate` are not being tested by this unit test. Can we enhance this test?

**Review by renormalize — COMMENTED** [review]
thanks for the PR @gflarity!
